File: LVJ_AdminVersion/ModelosPersistencia/Persistencia.cs
using System.Text.Json;

class Persistencia<T>
{
    protected List<T> objetos = new List<T>();
    protected string caminho;

    public Persistencia(string caminho) {this.caminho = caminho;}

    public void Inserir(T objeto)
    {
        // Abrir o Arquivo .json com Objetos da Lista de Objetos
        Abrir();
        // Obter o maior ID da Lista de Objetos e somar +1
        int id = objetos.Count == 0 ? 0 : objetos.Max(x => (int)x.GetType().GetProperty("id").GetValue(x));
        var idObjeto = objeto.GetType().GetProperty("id");
        if(idObjeto != null) idObjeto.SetValue(objeto, id + 1);
        // Adicionar o Objeto na Lista de Objetos e Salvar
        objetos.Add(objeto);
        Salvar();
    }

    public void Remover(T objeto)
    {
        // Remover o Objeto da Lista de Objetos e Salvar
        if(objeto != null)
        {
            objetos.Remove(objeto);
            Salvar();
        }
    }

    public void Atualizar(T objeto)
    {
        // Atualizar as Propriedades do Objeto da Lista de Objetos e Salvar
        int id = (int)objeto.GetType().GetProperty("id").GetValue(objeto);
        var x = ListarId(id);

        if(x != null)
        {
            int index = objetos.IndexOf(x);
            objetos[index] = objeto;
            Salvar();
        }
    }

    public List<T> Listar()
    {
        // Retorna a Lista de Objetos
        Abrir();
        return objetos;
    }

    public T ListarId(int id)
    {
        // Retornar o Objeto que possui o ID informado
        Abrir();
        return objetos.SingleOrDefault(x => (int)x.GetType().GetProperty("id").GetValue(x) == id);
    }

    public void Abrir()
    {
        // Abrir ou Criar um Arquivo .json
        objetos.Clear();
        try
        {
            // Abrir o Arquivo .json
            string adicionar = File.ReadAllText(caminho);
            objetos = JsonSerializer.Deserialize<List<T>>(adicionar);
        } catch (FileNotFoundException) {} catch (Exception) {}
    }

    public void Salvar()
    {
        try
        {
            // Salvar os Dados no Arquivo .json
            string salvar = JsonSerializer.Serialize<List<T>>(objetos);
            File.WriteAllText(caminho, salvar);
        } catch (Exception) {}
    } 
}

// Classes que possuem Herança com a Classe Persistencia

class Usuarios : Persistencia<Usuario>
{
    public Usuarios() : base(Path.Combine(Directory.GetParent(Directory.GetCurrentDirectory()).FullName, "Banco_Dados", "lista_usuarios.json")) {}
}

class Categorias : Persistencia<Categoria>
{
    public Categorias() : base(Path.Combine(Directory.GetParent(Directory.GetCurrentDirectory()).FullName, "Banco_Dados", "lista_categorias.json")) {}
}

class Produtos : Persistencia<Produto>
{
    public Produtos() : base(Path.Combine(Directory.GetParent(Directory.GetCurrentDirectory()).FullName, "Banco_Dados", "lista_produtos.json")) {}
}

class ProdutosVendas : Persistencia<ProdutoVenda>
{
    public ProdutosVendas() : base(Path.Combine(Directory.GetParent(Directory.GetCurrentDirectory()).FullName, "Banco_Dados", "lista_produtos_vendas.json")) {}
}

class Vendas : Persistencia<Venda>
{
    public Vendas() : base(Path.Combine(Directory.GetParent(Directory.GetCurrentDirectory()).FullName, "Banco_Dados", "lista_vendas.json")) {}
}

class FormasPagamento : Persistencia<FormaPagamento>
{
    public FormasPagamento() : base(Path.Combine(Directory.GetParent(Directory.GetCurrentDirectory()).FullName, "Banco_Dados", "lista_formas_pagamento.json")) {}
}

File: LVJ_ClienteVersion/ModeloPersistênciaPython/Persistencia.py
# using System.Text.Json;
import json
import os
from abc import ABC, abstractmethod
from typing import List, TypeVar

T = TypeVar('T')  # Define um tipo genérico T

class Persistencia(ABC):
    def __init__(self):
        self.objetos: List[T] = []

    def inserir(self, objeto: T):
        # Abrir o arquivo .json com objetos da lista de objetos
        self.abrir()
        # Obter o maior ID da lista de objetos e somar +1
        id = 0 if not self.objetos else max(getattr(x, 'id') for x in self.objetos)
        setattr(objeto, 'id', id + 1)  # Define o novo ID
        # Adicionar o objeto na lista de objetos e salvar
        self.objetos.append(objeto)
        self.salvar()

    def remover(self, objeto: T):
        # Remover o objeto da lista de objetos e salvar
        if objeto is not None:
            self.objetos.remove(objeto)
            self.salvar()

    def atualizar(self, objeto: T):
        # Atualizar as propriedades do objeto da lista de objetos e salvar
        id = getattr(objeto, 'id')
        x = self.listar_id(id)

        if x is not None:
            index = self.objetos.index(x)
            self.objetos[index] = objeto
            self.salvar()

    def listar(self) -> List[T]:
        # Retorna a lista de objetos
        self.abrir()
        return self.objetos

    def listar_id(self, id: int) -> T:
        # Retornar o objeto que possui o ID informado
        self.abrir()
        return next((x for x in self.objetos if getattr(x, 'id') == id), None)

    @abstractmethod
    def abrir(self):
        pass

    @abstractmethod
    def salvar(self):
        pass
